sources/osgAnimation/BasicAnimationManager.js

```diff
@@ -160,30 +148,25 @@ define( [
             this.addAnimations( animations );
         },
 
+
+        // push all animations into the queue
         addAnimations: function ( animations ) {
 
-            // instanceAnimation added
             var instanceAnimationList = this._addAnimation( animations );
 
-            // compute a map and set a targetID for each InstanceChannel from
-            // instanceAnimationList
-            var newTargetAdded = Animation.registerChannelTargetID( instanceAnimationList, this._targets, this._targetsMap );
-
-            // register targetID per type
-            for ( var i = 0; i < newTargetAdded.length; i++ ) {
-                var type = newTargetAdded[ i ].type;
-                this._targetID.push( createTargetID( i, ResultType[ type ].create() ) );
-            }
-
+            // qeue them to assign target
+            Array.prototype.push.apply( this._animationsToRegister, instanceAnimationList );
             this._dirty = true;
+            this._registerAnimations();
         },
 
+
         update: function ( node, nv ) {
 
             if ( this._dirty ) {
                 this._findAnimationUpdateCallback( node );
-                this._assignTargetToAnimationCallback();
-                this._dirty = false;
+                this._registerTargetFoundInAnimationCallback();
+                this._registerAnimations();
```

Comment:
Ok.

Still Seems risky to me,  dirty saying multiple things (are callback found, are target found in call back, are animations registered), and need correct order. As that code block is only to executed when dirty is true and no "_registerAnimations" has been called prior to the "update" call. 
Not that it does happens now, but code evolution might make the '_registerAnimation' be called prior an "update" call happens afterward, and those findAnimationCallback and _registerTargetFoundInAnimationCallback be not ever called once.